src/rich-text-editor.component.js

```diff
@@ -2,51 +2,42 @@ import React, {useState, useEffect, useRef, useContext, forwardRef} from 'react'
 import {RichTextContext} from './rich-text-container.component.js'
 
 const noop = () => {}
+const noopWithReturn = value => value;
 let globalBandicootId = 0
 
 export const RichTextEditor = forwardRef((props, editorRef) => {
+  const richTextContext = useContext(RichTextContext)
+  richTextContext.sanitizeHTML = props.sanitizeHTML;
```

Comment:
This makes it impossible for the user to change the sanitizeHTML prop after the initial render. See https://overreacted.io/writing-resilient-components/#dont-stop-the-data-flow-in-rendering

Instead:
```js
useEffect(() => {
  richTextContext.sanitizeHTML = props.sanitizeHTML
}, [props.sanitizeHTML])
```

Reply:
Doesn't the original implementation set the richTextContext.sanitizeHTML to the props.sanitizeHTML function _every time_ the component re-renders (for any reason)?  Wouldn't your suggestion primarily limit it to only set the props.sanitizeHTML function _IF_ the props.sanitizeHTML specifically changes.  I'm actually okay with limiting the times it is set (and wanted to), but was concerned that putting it in a useEffect would create an unexpected sequence issue where the useEffect could run _AFTER_ something else that was expecting the sanitizer to already be setup had done their things (thus meaning the html wouldn't be sanitized as expected).  My thought on the original implementation was to get the sanitizing function set as soon as it possibly could.  Maybe putting it in a useLayoutEffect would work though?  My other thought was to put it as a property on the RichTextContainer itself (again with a useLayoutEffect), but that felt a bit less natural.  Or maybe bandicoot needs some bootstrapping 'lifecycles', where certain things are guaranteed to happen before other things.  Thoughts? Does any of this make sense?

Reply:
ohhh i see now. I just missed this. Yeah you are right -- it's getting set every time. So no code change is needed here.